src/core/processing-queue.js

```diff
@@ -23,67 +22,83 @@ let priorityCount = 0;
 let unitSampler;
 
 /**
- * Advances the ProcessingQueue to the next item if it is ready.
- * @param {Boolean} last
+ * Advances the taskQueue to the next task. If the taskQueue is empty,
+ * process the testQueue
  */
 function advance() {
+	advanceTaskQueue();
+
+	if ( !config.queue.length ) {
+		advanceTestQueue();
+	}
+}
+
+/**
+ * Advances the taskQueue to the next task if it is ready and not empty.
+ */
+function advanceTaskQueue() {
 	const start = now();
 	config.depth = ( config.depth || 0 ) + 1;
 
 	while ( config.queue.length && !config.blocking ) {
 		const elapsedTime = now() - start;
 
 		if ( !defined.setTimeout || config.updateRate <= 0 || elapsedTime < config.updateRate ) {
-			if ( priorityCount > 0 ) {
```

Comment:
Was this code unneeded?  What was its original purpose? Can you explain why we don't need it now?

Reply:
In the original code, priorityCount's main purpose is to prioritize tests that was flagged to be executed first. The priorityCount was also incremented/decremented in `advance()` originally because if a tests needed to be inserted to the queue, it will be inserted after the tasks(which gets added to front of the queue) and after the already prioritized tests.
This is not necessary anymore since the task is in it's own queue, so we do not need to deal with the priortyCount when dealing with the tasks. (Only needed for the testQueue in the new implementation)